lib/pulsar/extended_interactor.rb

```diff
@@ -0,0 +1,37 @@
+module Pulsar
```

Comment:
This file hurts my brain 😅 

I'd change the `validate_context_for` method because it's duplicate and kind of strange to read a method that calls a method with the same name.

I also think maybe we can try to reduce the amount of metaprogramming and make it more straight forward.

Instead of changing how the `Interactor` works by adding modules we should include additional modules in each interactor. In this way we could have smaller modules and include only what's needed. For instance this module could be called `Pulsar::InteractorValidations`.